app/models/concerns/rdv/updatable.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Rdv::Updatable
+  extend ActiveSupport::Concern
+
+  def update_and_notify(author, attributes)
+    assign_attributes(attributes)
+    save_and_notify(author)
+  end
+
+  def save_and_notify(author)
+    Rdv.transaction do
+      self.updated_at = Time.zone.now
+
+      if status_changed?
+        self.cancelled_at = status.in?(%w[excused revoked noshow]) ? Time.zone.now : nil
+      end
+
+      previous_participations = rdvs_users.select(&:persisted?)
+
+      if save
+        notify!(author, previous_participations)
+        true
+      else
+        false
+      end
+    end
+  end
+
+  def rdv_user_token(user_id)
+    @rdv_users_tokens_by_user_id&.fetch(user_id, nil)
+  end
+
+  def notify!(author, previous_participations)
```

Comment:
quick question : la succession de `if .. end` sous-entend que dans certains use case, on peut envoyer 4 notifications ?

ce qui m'interroge, même si le max de notifs est de 2, ne préfère-t-on pas une seule notification qui synthétise les infos du type : "Votre RDV avec X du YY a bien été déplacé au ZZ"

Reply:
Merci pour les explications. Si j'ai bien suivi, le refacto n'est pas le sujet, mais plus le passage d'un service à un concern. So good for me. Tu merges ?